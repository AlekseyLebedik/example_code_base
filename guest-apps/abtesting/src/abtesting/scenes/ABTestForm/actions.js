import get from 'lodash/get';
import set from 'lodash/set';

import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';
import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';

import * as AT from './actionTypes';

export function fetchContextFailed(err) {
  return dispatch => {
    dispatch(GlobalSnackBarActions.show(err.response.data.error.msg, 'error'));
  };
}

export function fetchSegments(context) {
  return {
    type: AT.FETCH_SEGMENTS,
    context,
  };
}

export function fetchCategories(context) {
  return {
    type: AT.FETCH_CATEGORIES,
    context,
  };
}

export function fetchSegmentsSuccess(data, context) {
  return {
    type: AT.FETCH_SEGMENTS_SUCCESS,
    segments: data.data,
    context,
  };
}

export function fetchSegmentsFailed(err, context) {
  return dispatch => {
    dispatch({ type: AT.FETCH_SEGMENTS_FAILED, context });
    dispatch(GlobalSnackBarActions.show(err.response.data.error.msg, 'error'));
  };
}

export function deleteSegments(context, segmentIDs) {
  return {
    type: AT.DELETE_SEGMENTS,
    context,
    segmentIDs,
  };
}

export function deleteSegmentsSuccess(context, segmentIDs) {
  return {
    type: AT.DELETE_SEGMENTS_SUCCESS,
    context,
    segmentIDs,
  };
}

export function deleteSegmentsFailed(err) {
  return dispatch => {
    dispatch(nonCriticalHTTPError(err));
  };
}

export function fetchCategoriesSuccess(data) {
  return {
    type: AT.FETCH_CATEGORIES_SUCCESS,
    categories: data.data,
    context: data.context,
  };
}

export function fetchCategoriesFailed(err) {
  return dispatch => dispatch(nonCriticalHTTPError(err, true));
}

export function fetchFirstParties(context) {
  return {
    type: AT.FETCH_FIRST_PARTIES,
    context,
  };
}

export function fetchFirstPartiesSuccess(data) {
  return {
    type: AT.FETCH_FIRST_PARTIES_SUCCESS,
    firstParties: data.data,
    context: data.context,
  };
}

export function fetchFirstPartiesFailed(err) {
  return dispatch => dispatch(nonCriticalHTTPError(err, true));
}

export function fetchConfigs(context) {
  return {
    type: AT.FETCH_CONFIGS,
    context,
  };
}

export function fetchConfigsSuccess(data, context) {
  return {
    type: AT.FETCH_CONFIGS_SUCCESS,
    configs: data.data,
    context,
  };
}

export function fetchConfigsFailed(err, context) {
  return dispatch => {
    dispatch({ type: AT.FETCH_CONFIGS_FAILED, context });
    dispatch(GlobalSnackBarActions.show(err.response.data.error.msg, 'error'));
  };
}

export function saveConfig(values, context, resolve, reject, onAdd) {
  return {
    type: AT.SAVE_CONFIG,
    values,
    context,
    resolve,
    reject,
    onAdd,
  };
}

export function notifyCopyOfConfigurations(previousConfig) {
  return dispatch =>
    dispatch(
      GlobalSnackBarActions.show(
        `Creating a new config cloned from ${previousConfig.name} to this context.`,
        'info'
      )
    );
}

export function saveConfigSuccess(context, values, update = false) {
  return {
    type: !update ? AT.ADD_CONFIG_SUCCESS : AT.UPDATE_CONFIG_SUCCESS,
    context,
    values,
  };
}

export function saveConfigFailed(err) {
  return dispatch => dispatch(nonCriticalHTTPError(err, true));
}

export function saveTest(
  data,
  history,
  resolve,
  reject,
  testID,
  newCohorts,
  nonMatchingConfigs
) {
  return {
    type: AT.SAVE_TEST,
    data,
    history,
    resolve,
    reject,
    testID,
    newCohorts,
    nonMatchingConfigs,
  };
}

export function saveTestSuccess(history) {
  return dispatch => {
    dispatch({ type: AT.SAVE_TEST_SUCCESS });
    dispatch(GlobalSnackBarActions.show('Test saved successfully.', 'success'));
    history.push('/abtesting/schedule');
  };
}

export function saveTestFailed(err) {
  return dispatch => {
    dispatch({ type: AT.SAVE_TEST_FAILED });
    dispatch(GlobalSnackBarActions.show(err.response.data.error.msg, 'error'));
  };
}

export function uploadManualSourceFileFailed(
  err,
  history,
  titleID,
  environment,
  testID
) {
  return dispatch => {
    const msg = get(err, 'response.data.error.invalid[0].msg');
    if (msg) set(err, 'response.data.error.msg', msg);
    dispatch(
      GlobalSnackBarActions.show(
        `Test saved successfully. Manual cohort (user IDs) file upload failed with error: ${msg}`,
        'error'
      )
    );
    history.push(`/abtesting/edit/${titleID}/${environment}/${testID}`);
  };
}

export function fetchCohortUsers(context, testID, cohortID) {
  return {
    type: AT.FETCH_COHORT_USERS,
    context,
    testID,
    cohortID,
  };
}

export function fetchCohortUsersSuccess(data, cohortID) {
  return {
    type: AT.FETCH_COHORT_USERS_SUCCESS,
    users: data.data,
    cohortID,
  };
}

export function fetchCohortUsersFailed(err, cohortID) {
  return dispatch => {
    dispatch({ type: AT.FETCH_COHORT_USERS_FAILED, cohortID });
    dispatch(GlobalSnackBarActions.show(err.response.data.error.msg, 'error'));
  };
}

export function deleteCohortUsers(context, testID, cohortID) {
  return {
    type: AT.DELETE_COHORT_USERS,
    context,
    testID,
    cohortID,
  };
}

export function deleteCohortUsersSuccess(cohortID) {
  return {
    type: AT.DELETE_COHORT_USERS_SUCCESS,
    cohortID,
  };
}

export function deleteCohortUsersFailed(err) {
  return dispatch => {
    dispatch(GlobalSnackBarActions.show(err.response.data.error.msg, 'error'));
  };
}
