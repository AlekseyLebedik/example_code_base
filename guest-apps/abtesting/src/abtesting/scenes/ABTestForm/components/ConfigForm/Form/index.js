import React from 'react';
import PropTypes from 'prop-types';
import { Field, Form, reduxForm } from 'redux-form';

import Input from 'dw/core/components/FormFields/Input';
import NonFieldError from 'dw/core/components/FormFields/NonFieldError';
import * as V from 'dw/core/components/FormFields/validation';

import styles from './index.module.css';

const ConfigForm = props => {
  const { handleSubmit, externalSubmit } = props;
  return (
    <div className={styles.container}>
      <Form onSubmit={handleSubmit(externalSubmit)}>
        <Field name="error" component={NonFieldError} />
        <Field
          name="name"
          component={Input}
          label="Name"
          fullWidth
          validate={[V.required]}
        />
        <Field
          name="serviceID"
          component={Input}
          label="Target"
          fullWidth
          validate={[V.required]}
        />
        <Field
          name="modifiers"
          component={Input}
          label="Custom config (JSON structure)"
          fullWidth
          multiline
          rows={4}
        />
      </Form>
    </div>
  );
};

ConfigForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  externalSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  enableReinitialize: true,
  keepDirtyOnReinitialize: true,
})(ConfigForm);
