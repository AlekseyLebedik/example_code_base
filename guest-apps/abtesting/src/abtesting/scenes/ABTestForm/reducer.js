import * as AT from './actionTypes';

const INITIAL_STATE = {
  cohortUsers: {},
  configs: {},
  segments: {},
  contexts: {},
  saving: false,
  cohorts: [],
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.FETCH_SEGMENTS:
      return {
        ...state,
        segments: { ...state.segments, [action.context]: null },
      };
    case AT.FETCH_SEGMENTS_SUCCESS:
      return {
        ...state,
        segments: { ...state.segments, [action.context]: action.segments },
      };
    case AT.FETCH_SEGMENTS_FAILED:
      return {
        ...state,
        segments: { ...state.segments, [action.context]: [] },
      };
    case AT.DELETE_SEGMENTS_SUCCESS:
      return {
        ...state,
        segments: {
          ...state.segments,
          [action.context]: state.segments[action.context].filter(
            item => !action.segmentIDs.includes(item.segmentID)
          ),
        },
      };
    case AT.FETCH_COHORT_USERS:
      return {
        ...state,
        cohortUsers: { ...state.cohortUsers, [action.cohortID]: null },
      };
    case AT.FETCH_COHORT_USERS_SUCCESS:
      return {
        ...state,
        cohortUsers: { ...state.cohortUsers, [action.cohortID]: action.users },
      };
    case AT.FETCH_COHORT_USERS_FAILED:
    case AT.DELETE_COHORT_USERS_SUCCESS:
      return {
        ...state,
        cohortUsers: { ...state.cohortUsers, [action.cohortID]: [] },
      };
    case AT.FETCH_CONFIGS:
      return {
        ...state,
        configs: { ...state.configs, [action.context]: null },
      };
    case AT.FETCH_CATEGORIES_SUCCESS:
      return {
        ...state,
        contexts: {
          ...state.contexts,
          [action.context]: {
            ...(state.contexts[action.context] || {}),
            categories: action.categories,
          },
        },
      };
    case AT.FETCH_FIRST_PARTIES_SUCCESS:
      return {
        ...state,
        contexts: {
          ...state.contexts,
          [action.context]: {
            ...(state.contexts[action.context] || {}),
            firstParties: action.firstParties,
          },
        },
      };
    case AT.FETCH_CONFIGS_SUCCESS:
      return {
        ...state,
        configs: { ...state.configs, [action.context]: action.configs },
      };
    case AT.FETCH_CONFIGS_FAILED:
      return {
        ...state,
        configs: { ...state.configs, [action.context]: [] },
      };
    case AT.ADD_CONFIG_SUCCESS:
      return {
        ...state,
        configs: {
          ...state.configs,
          [action.context]: [
            ...state.configs[action.context],
            {
              configID: action.values.configID,
              name: action.values.name,
              created: Date.now() / 1000,
              updated: Date.now() / 1000,
              serviceID: action.values.serviceID,
              modifiers: action.values.modifiers,
            },
          ],
        },
      };
    case AT.UPDATE_CONFIG_SUCCESS:
      return {
        ...state,
        configs: {
          ...state.configs,
          [action.context]: state.configs[action.context].map(c =>
            c.configID === action.values.configID
              ? { ...c, ...action.values }
              : c
          ),
        },
      };
    case AT.SAVE_TEST:
      return {
        ...state,
        saving: true,
      };
    case AT.SAVE_TEST_SUCCESS:
    case AT.SAVE_TEST_FAILED:
      return { ...state, saving: false };
    default:
      return state;
  }
};
