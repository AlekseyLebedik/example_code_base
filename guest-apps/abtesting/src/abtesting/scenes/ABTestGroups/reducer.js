import { combineReducers } from 'redux';
import reduceReducers from 'reduce-reducers';

import {
  createFetchReducer,
  INITIAL_STATE,
} from '@demonware/devzone-core/helpers/reducers';
import {
  GROUPS_LIST_PREFIX,
  GROUPS_DETAILS_PREFIX,
  GROUPS_DETAILS_TESTS_PREFIX,
  GROUPS_DETAILS_CONFIGS_PREFIX,
} from './constants';

const DETAILS_INITIAL_STATE = {
  loading: false,
  members: [],
};

const fetchGroupsReducer = createFetchReducer(GROUPS_LIST_PREFIX);
const fetchTestsReducer = createFetchReducer(GROUPS_DETAILS_TESTS_PREFIX);
const fetchConfigsReducer = createFetchReducer(GROUPS_DETAILS_CONFIGS_PREFIX);

const groupDetailsReducer = (state = DETAILS_INITIAL_STATE, action) => {
  switch (action.type) {
    case `${GROUPS_DETAILS_PREFIX}_FETCH`: {
      return {
        ...state,
        loading: true,
        groupID: action.params.groupID,
        members: [],
      };
    }
    case `${GROUPS_DETAILS_PREFIX}_FETCH_SUCCESS`:
      return state.groupID === action.data.groupID
        ? {
            ...state,
            ...action.data,
            loading: false,
          }
        : state;
    case `${GROUPS_DETAILS_PREFIX}_FETCH_FAILED`:
      return {
        ...state,
        loading: false,
      };
    case `${GROUPS_DETAILS_PREFIX}_ADD_GROUP_MEMBER_SUCCESS`: {
      const userIds = state.members.map(user => user.userID);
      const users = action.user.filter(user => !userIds.includes(user.userID));
      return state.groupID === action.groupID && users.length > 0
        ? {
            ...state,
            members: [...users, ...state.members],
          }
        : state;
    }
    case `${GROUPS_DETAILS_PREFIX}_REMOVE_GROUP_MEMBERS_SUCCESS`: {
      const ids = action.users.map(u => u.userID);
      return state.groupID === action.groupID &&
        state.members.some(u => ids.includes(u.userID))
        ? {
            ...state,
            members: state.members.filter(u => !ids.includes(u.userID)),
          }
        : state;
    }
    default:
      return state;
  }
};

const deleteReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case `${GROUPS_LIST_PREFIX}_DELETE_GROUP_SUCCESS`: {
      return {
        ...state,
        data: action.group
          ? state.data.filter(item => item !== action.group)
          : state.data,
        loading: false,
      };
    }
    default:
      return state;
  }
};

export default combineReducers({
  groups: reduceReducers(fetchGroupsReducer, deleteReducer, INITIAL_STATE),
  groupDetails: groupDetailsReducer,
  tests: fetchTestsReducer,
  configs: fetchConfigsReducer,
});
