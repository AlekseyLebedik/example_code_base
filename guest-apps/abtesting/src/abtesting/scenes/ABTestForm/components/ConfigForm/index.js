import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { submit } from 'redux-form';

import { hasData } from 'dw/core/helpers/object';
import ModalHandlers from 'dw/core/components/ModalHandlers';

import * as actions from '../../actions';
import { selectedContextFormSelector } from '../../selectors';
import ConfigModal from './Modal';

const stateToProps = (state, props) => {
  const { formName } = props;
  return {
    selectedContext: selectedContextFormSelector(state),
    isAddConfigModalVisible: ModalHandlers.isVisibleSelector(state, formName),
    formName,
  };
};

const dispatchToProps = (dispatch, props) => {
  const { formName } = props;
  return {
    closeAddConfigModal: () => dispatch(ModalHandlers.close(formName)),

    handlerAddConfig: () => dispatch(submit(formName)),
    saveConfig: (values, context, onAdd) =>
      new Promise((_, reject) =>
        dispatch(
          actions.saveConfig(
            values,
            context,
            () => ModalHandlers.close(formName),
            reject,
            onAdd
          )
        )
      ),
  };
};

const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  ...stateProps,
  ...dispatchProps,
  ...ownProps,
  saveConfig: values =>
    dispatchProps.saveConfig(
      values,
      stateProps.selectedContext,
      ownProps.onAdd
    ),
});

const ConfigForm = props => {
  const {
    closeAddConfigModal,
    handlerAddConfig,
    isAddConfigModalVisible,
    formName,
    saveConfig,
    config,
    cohortName,
  } = props;

  return (
    <ConfigModal
      visible={isAddConfigModalVisible}
      onCancel={closeAddConfigModal}
      onRemoteSubmit={handlerAddConfig}
      onSubmit={saveConfig}
      formName={formName}
      config={!hasData(config) && cohortName ? { name: cohortName } : config}
    />
  );
};

ConfigForm.propTypes = {
  closeAddConfigModal: PropTypes.func.isRequired,
  handlerAddConfig: PropTypes.func.isRequired,
  isAddConfigModalVisible: PropTypes.bool.isRequired,
  formName: PropTypes.string.isRequired,
  saveConfig: PropTypes.func.isRequired,
  config: PropTypes.object,
  cohortName: PropTypes.string,
};

ConfigForm.defaultProps = {
  config: {},
  cohortName: undefined,
};

export default connect(stateToProps, dispatchToProps, mergeProps)(ConfigForm);
