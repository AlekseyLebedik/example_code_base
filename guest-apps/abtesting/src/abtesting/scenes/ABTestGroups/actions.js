import get from 'lodash/get';
import set from 'lodash/set';

import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';
import { createFetch } from '@demonware/devzone-core/helpers/actions';

import {
  GROUPS_LIST_PREFIX,
  GROUPS_DETAILS_PREFIX,
  GROUPS_DETAILS_TESTS_PREFIX,
  GROUPS_DETAILS_CONFIGS_PREFIX,
  GROUPS_SERVICE,
} from './constants';

export const fetchGroups = params =>
  createFetch(GROUPS_LIST_PREFIX, null, {
    ...params,
    service: GROUPS_SERVICE,
    bypassContextValidation: 1,
  });

export const getGroupDetails = (groupID, context) =>
  createFetch(GROUPS_DETAILS_PREFIX, null, {
    groupID,
    context,
    service: GROUPS_SERVICE,
    bypassContextValidation: 1,
  });

export const fetchTests = context =>
  createFetch(GROUPS_DETAILS_TESTS_PREFIX, null, { context });

export const fetchConfigs = context =>
  createFetch(GROUPS_DETAILS_CONFIGS_PREFIX, null, { context });

export const createGroup = (group, context) => ({
  type: `${GROUPS_DETAILS_PREFIX}_CREATE_GROUP`,
  group,
  context,
  service: GROUPS_SERVICE,
  bypassContextValidation: 1,
});

export const createGroupSuccess = group => ({
  type: `${GROUPS_DETAILS_PREFIX}_CREATE_OBJECT_GROUP_SUCCESS`,
  group,
});

export const addGroupMember = (groupID, user, context) => ({
  type: `${GROUPS_DETAILS_PREFIX}_ADD_GROUP_MEMBER`,
  groupID,
  user,
  context,
  service: GROUPS_SERVICE,
  bypassContextValidation: 1,
});

export const addGroupMemberSuccess = (groupID, user) => ({
  type: `${GROUPS_DETAILS_PREFIX}_ADD_GROUP_MEMBER_SUCCESS`,
  groupID,
  user: Array.isArray(user) ? user : [user],
});

export const removeGroupMembers = (groupID, users, context) => ({
  type: `${GROUPS_DETAILS_PREFIX}_REMOVE_GROUP_MEMBERS`,
  groupID,
  users,
  context,
  service: GROUPS_SERVICE,
  bypassContextValidation: 1,
});

export const removeGroupMembersSuccess = (groupID, users) => ({
  type: `${GROUPS_DETAILS_PREFIX}_REMOVE_GROUP_MEMBERS_SUCCESS`,
  groupID,
  users,
});

export const replaceUsers = (groupId, values, context) => ({
  type: `${GROUPS_DETAILS_PREFIX}_REPLACE_USERS`,
  groupId,
  values,
  context,
  service: GROUPS_SERVICE,
  bypassContextValidation: 1,
});

export const replaceUsersSuccess = () => dispatch => {
  dispatch(GlobalSnackBarActions.show(`Users were replaced.`, 'success'));
  dispatch({
    type: `${GROUPS_DETAILS_PREFIX}_REPLACE_USERS_SUCCESS`,
  });
};

export const deleteGroup = (group, context) => ({
  type: `${GROUPS_LIST_PREFIX}_DELETE_GROUP`,
  group,
  context,
  service: GROUPS_SERVICE,
  bypassContextValidation: 1,
});

export const deleteGroupSuccess = group => dispatch => {
  dispatch(
    GlobalSnackBarActions.show(
      `Successfully deleted group with name ${group.groupName}.`
    ),
    'success'
  );

  dispatch({
    type: `${GROUPS_LIST_PREFIX}_DELETE_GROUP_SUCCESS`,
    group,
  });
};

export const handleActionFailed = err => {
  const msg = get(err, 'response.data.error.invalid[0].msg');
  if (msg) set(err, 'response.data.error.msg', msg);
  return nonCriticalHTTPError(err);
};
