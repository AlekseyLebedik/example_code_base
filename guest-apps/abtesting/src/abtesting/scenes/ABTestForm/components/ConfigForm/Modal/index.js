import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Dialog from 'dw/core/components/Dialog';

import { hasData } from 'dw/core/helpers/object';

import Form from '../Form';
import styles from './index.module.css';

const ConfigForm = props => {
  const { visible, onCancel, onRemoteSubmit, onSubmit, formName, config } =
    props;
  const hasConfig = hasData(config);

  const footerButtons = [
    <Button key="cancel" label="Cancel" onClick={onCancel}>
      Cancel
    </Button>,
    <Button key="update" color="primary" onClick={onRemoteSubmit}>
      {hasConfig ? 'Update Config' : 'Add Config'}
    </Button>,
  ];

  return (
    <div className={styles.modalContainer}>
      <Dialog
        title={hasConfig ? 'Update Config' : 'Add Config'}
        actions={footerButtons}
        modal
        open={visible}
        onRequestClose={onCancel}
        contentStyle={{ width: '500px' }}
      >
        <Form
          externalSubmit={onSubmit}
          form={formName}
          initialValues={config}
        />
      </Dialog>
    </div>
  );
};

ConfigForm.propTypes = {
  visible: PropTypes.bool.isRequired,
  onCancel: PropTypes.func.isRequired,
  onRemoteSubmit: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  formName: PropTypes.string.isRequired,
  config: PropTypes.object,
};

ConfigForm.defaultProps = {
  config: {},
};

export default ConfigForm;
