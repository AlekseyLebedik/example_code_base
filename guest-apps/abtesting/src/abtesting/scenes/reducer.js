import { combineReducers } from 'redux';

import { reducer as ABTestFormReducer } from './ABTestForm/reducer';
import { reducer as UpdateReducer } from './Update/reducer';
import ABTestGroupsReducer from './ABTestGroups/reducer';

const reducer = combineReducers({
  ABTestForm: ABTestFormReducer,
  Update: UpdateReducer,
  ABTestGroups: ABTestGroupsReducer,
});

export default reducer;
