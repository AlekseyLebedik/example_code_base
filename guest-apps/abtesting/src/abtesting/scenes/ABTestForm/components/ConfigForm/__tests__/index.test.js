import React from 'react';
import { createTestStore, shallowUntilTarget } from 'dw/test-utils';

import reducer from 'dw/core/components/ModalHandlers/reducer';
import { openModalHandler } from 'dw/core/components/ModalHandlers/actions';
import ConfigForm from '../index';

describe('ABTesting component ConfigForm', () => {
  let store = null;
  beforeEach(() => {
    ({ store } = createTestStore('Core.ModalHandlers', reducer));
  });

  it('renders default form', () => {
    const props = {
      store,
      formName: 'Test',
    };
    expect(
      shallowUntilTarget(<ConfigForm {...props} />, 'ConfigForm')
    ).toMatchSnapshot();
  });

  it('renders visible form', () => {
    const formName = 'test';
    const props = {
      store,
      formName,
      config: {
        updated: '1542027298',
        modifiers: '{"second_campaign_value": 2}',
        name: 'cfg1',
        created: '1542027282',
        serviceID: 't8',
        context: 'game2',
        configID: '15716207803415653925',
        immutable: true,
      },
    };
    store.dispatch(openModalHandler(formName));
    expect(
      shallowUntilTarget(<ConfigForm {...props} />, 'ConfigForm')
    ).toMatchSnapshot();
  });
});
