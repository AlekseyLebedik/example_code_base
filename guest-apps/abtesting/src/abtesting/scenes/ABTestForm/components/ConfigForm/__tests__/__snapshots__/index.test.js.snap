// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ABTesting component ConfigForm renders default form 1`] = `
<ConfigForm
  config={Object {}}
  formName="Test"
  onCancel={[Function]}
  onRemoteSubmit={[Function]}
  onSubmit={[Function]}
  visible={false}
/>
`;

exports[`ABTesting component ConfigForm renders visible form 1`] = `
<ConfigForm
  config={
    Object {
      "configID": "15716207803415653925",
      "context": "game2",
      "created": "1542027282",
      "immutable": true,
      "modifiers": "{\\"second_campaign_value\\": 2}",
      "name": "cfg1",
      "serviceID": "t8",
      "updated": "1542027298",
    }
  }
  formName="test"
  onCancel={[Function]}
  onRemoteSubmit={[Function]}
  onSubmit={[Function]}
  visible={true}
/>
`;
